code/modules/mob/living/living_defense.dm

```diff
@@ -406,6 +406,9 @@
 	if(HAS_TRAIT(src, TRAIT_NOFLASH))
 		return FALSE
 	if(get_eye_protection() < intensity && (affect_silicon || override_blindness_check || !is_blind()))
+		if(client?.prefs?.darkened_flash)
+			type = /atom/movable/screen/fullscreen/flash/black
```

Comment:
The `type` parameter never gets overridden, right?
This is a bad pattern we've started using, filling proc calls with arguments like these.
If you could remove it and just make it a proc variable, that'd be good.
It would imply changing the arguments on the 6 proc definitions, but thankfully there's no need to change any of the proc calls.

Reply:
It actually does get overridden for silicon's flash_act to use the static version, which seems reasonable to mme. I've tweaked the proc a little bit to make this a little less silly looking, but this lets us make sure we're overriding the flash type which I think is justified since it's an accessibility thing.